src/system/group.cr

```diff
@@ -0,0 +1,158 @@
+require "c/grp"
+require "c/sys/limits"
+
+struct System::Group
+  module Limits
+    {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+      GID_MAX = LibC::GID_MAX
+    {% elsif flag?(:linux) %}
+      GID_MAX = 0xffffffff_u32
+    {% else %}
+      GID_MAX = 0xffff_u32
+    {% end %}
+    NGROUPS_MAX = LibC::NGROUPS_MAX
+  end
+
+  alias GID = LibC::GidT
+
+  # Returns the group name for the given group ID.
+  # Raises `System::Group::NotFoundError` if no user exists.
+  def self.name(gid : Int) : String
```

Comment:
You should really just implement this as `name?(gid) || raise ...`

Reply:
Does that not add a crazy cost to calling the `x?` method? As the Error object has to be allocated, initialized, returned and discarded just to return a nil?

Reply:
No? I don't understand what you mean? I never suggested modifying the `?` method in any way.